test/src/main/java/org/apache/accumulo/test/functional/MemoryFreeingIterator.java

```diff
@@ -28,23 +28,39 @@
 import org.apache.accumulo.core.iterators.IteratorEnvironment;
 import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
 import org.apache.accumulo.core.iterators.WrappingIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 public class MemoryFreeingIterator extends WrappingIterator {
 
-  @Override
-  public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options,
-      IteratorEnvironment env) throws IOException {
-    super.init(source, options, env);
+  private static final Logger LOG = LoggerFactory.getLogger(MemoryFreeingIterator.class);
+
+  @SuppressFBWarnings(value = "DM_GC", justification = "gc is okay for test")
+  public MemoryFreeingIterator() {
+    super();
+    LOG.info("Freeing consumed memory");
     MemoryConsumingIterator.freeBuffers();
     while (this.isRunningLowOnMemory()) {
+      System.gc();
```

Comment:
Calling this may do nothing at all. Is that okay in this test?

Reply:
I agree that System.gc is a hint and may do nothing. The real thing that needs to occur is the call to `MemoryConsumingIterator.freeBuffers()`. The JVM GC should eventually run as the system tries to use more memory. Will it cause a flaky test? Unknown.

Reply:
See previous comment concerning the `System.gc` call in a loop, testing for memory recovery.